src/qilib/configuration_helper/adapters/ami430_instrument_adapter.py

```diff
@@ -48,40 +43,25 @@ def __init__(self, address: str, instrument_name: Optional[str] = None) -> None:
         self.field_variation_tolerance = 0.01
 
     def apply(self, config: PythonJsonStructure) -> None:
-        """ Does not apply config, except for instrument name, to device, but compares config to device settings.
+        """ Does not apply config only  compares config to device settings.
+
+        Configuration Parameter with setter commands are matched for equality, on mismatch
+        error is raised
 
         Args:
             config: Containing the instrument configuration.
-
-        Raises:
-            ConfigurationError: If config does not match device configuration or difference in field value is greater
-                than the field_variation_tolerance.
-
         """
-        device_config = self.read(True)
-        parameters = [parameter for parameter in config if hasattr(self._instrument.parameters[parameter], 'set')]
-        for parameter in parameters:
-            if parameter == 'field':
-                self._check_field_value(config[parameter]['value'], device_config[parameter]['value'])
-            elif 'value' in config[parameter]:
-                self._assert_value_matches(config[parameter]['value'], device_config[parameter]['value'], parameter)
-        if 'name' in config:
-            self._instrument.name = config['name']
+        super().apply(config)
 
     def _filter_parameters(self, parameters: PythonJsonStructure) -> PythonJsonStructure:
         for values in parameters.values():
             if 'val_mapping' in values:
                 values.pop('val_mapping')
         return parameters
 
-    def _check_field_value(self, config_value: float, device_value: float) -> None:
-        delta = math.fabs(config_value - device_value)
-        if delta > self.field_variation_tolerance:
-            raise ConfigurationError(
-                "Target value for field does not match device value: {}T != {}T".format(config_value, device_value))
-
-    @staticmethod
-    def _assert_value_matches(config_value: Any, device_value: Any, parameter: str) -> None:
-        if config_value != device_value:
-            raise ConfigurationError(
-                "Configuration for {} does not match: '{}' != '{}'".format(parameter, config_value, device_value))
+    def _compare_config_values(self, config_value: Any, device_value: Any, parameter: str = None) -> bool:
```

Comment:
type `str` can not be `None` by default. you can use  `Optional[str] = None`

Reply:
done